SOS/SERV/sos/sos_program.c

```diff
@@ -0,0 +1,621 @@
+/**
+ * @fileName	:	sos_program.c
+ * @Created		: 	2:08 AM Tuesday, May 30, 2023
+ * @Author		: 	Hamsters
+ * @brief		: 	Program file for Simple Operating System
+ */
+#include "sos_interface.h"
+#include "sos_private.h"
+static str_sos_task_t_*	gl_arr_ptr_str_task[SOS_NUMBER_OF_TASKS] = {NULL_PTR};
+uint32_t_ gl_uint32_tick_counter = 0; uint8_t_ gl_uint8_number_of_tasks_added = 0;
+enu_sos_scheduler_state_t_	gl_enu_sos_scheduler_state = SOS_SCHEDULER_UNINITIALIZED;
+/**
+*	@syntax				:	sos_init(void);
+*	@description		:	Initializes the sos module
+*	@Sync\Async      	:	Synchronous
+*  @Reentrancy      	:	Reentrant
+*  @Parameters (in) 	:	None
+*  @Parameters (out)	:	None
+*  @Return value		:	SOS_STATUS_SUCCESS in case timers doesn't return an error
+*							SOS_STATUS_INVALID_STATE in case timers return an error
+*/
+enu_sos_status_t_ sos_init(void)
+{
+    if (timer_init(  )==TIMER_NOK)

```

Comment:
You have to guard this function from being initialized multiple times. Try to include this inside your SOS state machine